File: 0001. Two Sum/0001.java
// O(n) time complexity
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hash = new HashMap<>();
        for(int i=0;i<nums.length;i++){
            int remaining = target - nums[i]; // nums[i] + remaining = target
            if(hash.containsKey(remaining))
                return new int []{hash.get(remaining), i};
            hash.put(nums[i], i);
        }
        return new int []{};
    }
}

File: 0001. Two Sum/0001.cpp
//O(N^2) solution
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> result;
        for (int i=0;i<nums.size()-1;i++){
            for (int j=i+1;j<nums.size();j++){
                if(nums[i]+nums[j]==target){ 
                    result = {i,j};
                    break;
                }
            }
        }
        return result;
    }
};